File: frontend/src/types/system.ts
export interface ModelInfo {
  name: string
  role: 'fast' | 'reasoning' | 'vision' | 'embedding'
  size: string
  status: 'loaded' | 'idle'
}

export interface ServiceHealth {
  name: string
  up: boolean
}

export interface ResourceUsage {
  ramUsedGb: number
  ramTotalGb: number
  gpuUsedGb: number
  gpuTotalGb: number
}

export interface SystemStatus {
  models: ModelInfo[]
  services: ServiceHealth[]
  resources: ResourceUsage
}

File: backend/app/schemas/system.py
from typing import Literal

from app.schemas.base import CamelModel


class ModelInfo(CamelModel):
    name: str
    role: Literal["fast", "reasoning", "vision", "embedding"]
    size: str
    status: Literal["loaded", "idle"]


class ServiceHealth(CamelModel):
    name: str
    up: bool


class ResourceUsage(CamelModel):
    ram_used_gb: float
    ram_total_gb: float
    gpu_used_gb: float
    gpu_total_gb: float


class SystemStatus(CamelModel):
    models: list[ModelInfo]
    services: list[ServiceHealth]
    resources: ResourceUsage